plugins/ufc-scoreboard/base_odds_manager.py

```
@@ -0,0 +1,284 @@
+            if odds_data:
+                self.cache_manager.set(cache_key, odds_data)
+                self.logger.info(f"Saved odds data to cache for {cache_key}")
+            else:
+                self.logger.debug(f"No odds data available for {cache_key}")
+                # Cache the fact that no odds are available to avoid repeated API calls
+                self.cache_manager.set(cache_key, {"no_odds": True})
+
+            return odds_data
+
+        except requests.exceptions.RequestException as e:
+            self.logger.error(f"Error fetching odds from ESPN API for {cache_key}: {e}")
```

Ruff (0.15.0)

[warning] Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)